processing/src/main/java/org/apache/druid/query/aggregation/first/DoubleFirstAggregatorFactory.java

```diff
@@ -125,6 +138,21 @@ public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
     }
   }
 
+  @Override
+  public VectorAggregator factorizeVector(
+      VectorColumnSelectorFactory columnSelectorFactory
+  )
+  {
+    ColumnCapabilities capabilities = columnSelectorFactory.getColumnCapabilities(fieldName);
+    if (capabilities != null && Types.isNumeric(capabilities)) {
```

Comment:
oops sorry for the confusion, `Types.isNumeric` includes the null check on `capabilities` so it isn't needed here